lib/auth/saml.go

```diff
@@ -148,7 +178,19 @@ func (a *Server) getSAMLProvider(conn types.SAMLConnector) (*saml2.SAMLServicePr
 	return serviceProvider, nil
 }
 
-func (a *Server) calculateSAMLUser(connector types.SAMLConnector, assertionInfo saml2.AssertionInfo, request *services.SAMLAuthRequest) (*createUserParams, error) {
+func (a *Server) ssoDiagInfo(ctx context.Context, authKind string, id string, infoType types.SSOInfoType, value interface{}) {
```

Comment:
Nit: `ssoDiagInfo` reads more like a variable than a function name. It's not clear: does it return ssoDiagInfo, does it set it somewhere, etc.

Reply:
Done.